### PR TITLE
Parse reuse-grpc-port and reuse-http-port as booleans

### PR DESCRIPTION
**Pull request by dyastremsky**
The reuse-grpc-port and reuse-http-port flags are documented in the options as booleans but parsed as integers. This PR makes it so they are parsed as booleans.

Addresses [6481](https://github.com/triton-inference-server/server/issues/6481).

### CODE REVIEW

File: qa/L0_socket/test.sh [hunk 1]
```diff
@@ -357,9 +357,9 @@ for p in http grpc; do
     #   (a) Test default metrics-address being same as http-address
     #   (b) Test setting metrics-address explicitly to 0.0.0.0
     #   (c) Test setting metrics-address explicitly to 127.0.0.1
-    SERVER0_ARGS="--model-repository=$DATADIR --metrics-port 8002 --reuse-http-port=1 --reuse-grpc-port=1"
-    SERVER1_ARGS="--model-repository=$DATADIR --metrics-address 0.0.0.0 --metrics-port 8003 --reuse-http-port=1 --reuse-grpc-port=1"
-    SERVER2_ARGS="--model-repository=$DATADIR --metrics-address 127.0.0.2 --metrics-port 8004 --reuse-http-port=1 --reuse-grpc-port=1"
+    SERVER0_ARGS="--model-repository=$DATADIR --metrics-port 8002 --reuse-http-port=true --reuse-grpc-port=true"
+    SERVER1_ARGS="--model-repository=$DATADIR --metrics-address 0.0.0.0 --metrics-port 8003 --reuse-http-port=1 --reuse-grpc-port=true"
```
Comment (fpetrini15) [inline qa/L0_socket/test.sh:361]:
```suggestion
    SERVER1_ARGS="--model-repository=$DATADIR --metrics-address 0.0.0.0 --metrics-port 8003 --reuse-http-port=true --reuse-grpc-port=true"
```

File: qa/L0_socket/test.sh [hunk 2]
```diff
@@ -357,9 +357,9 @@ for p in http grpc; do
...
+    SERVER2_ARGS="--model-repository=$DATADIR --metrics-address 127.0.0.2 --metrics-port 8004 --reuse-http-port=1 --reuse-grpc-port=true"
```
Comment (fpetrini15) [inline qa/L0_socket/test.sh:362]:
```suggestion
    SERVER2_ARGS="--model-repository=$DATADIR --metrics-address 127.0.0.2 --metrics-port 8004 --reuse-http-port=true --reuse-grpc-port=true"
```

**Comment by kthui** [comment]
I wonder what you think about changing the docs instead? In case people are supplying an `int` instead of `bool` in the real world?

**Comment by dyastremsky** [comment]
> I wonder what you think about changing the docs instead? In case other people are silently supplying an `int` instead of `bool` in the real world?

For context, the docs are the ones [here](https://github.com/triton-inference-server/server/blob/e865b47d9e9e1beaca12c51fd6b6ab925bc8cd96/src/command_line_parser.cc#L453) (in the parser itself).

As far as real world usage, as you can see from the fact that the test was passing before the two commits Francesco just added that this should work with values 0 and 1 because they are accepted by [ParseOption](https://github.com/triton-inference-server/server/blob/e865b47d9e9e1beaca12c51fd6b6ab925bc8cd96/src/command_line_parser.cc#L168-L184). Beyond that, I think previously any non-0 was accepted as "true" and now any non-zero integer other than 1 would throw an error.

This is a good point, so CC'ing @nnshah1 and @GuanLuo for their thoughts.

**Review by kthui — APPROVED** [review]
>  I think previously any non-0 was accepted as "true" and now any non-zero integer other than 1 would throw an error.

Thanks for clarifying, I don't know how many people could fall into this "trap", other than that, LGTM

**Comment by nnshah1** [comment]
Good with the change - we can release note it to give people a heads up.

**Comment by dyastremsky** [comment]
Awesome, thank you Jacky and Neelay! I'll add it to the release notes.

**Pull request merged**